src/platform/src/main/java/org/geoserver/platform/resource/FileSystemResourceStore.java

```diff
@@ -162,7 +162,7 @@ public String toString() {
         File file;
 
         public FileSystemResource(String path) {
-            this.path = path;
+            this.path = Paths.convert(path);
```

Comment:
So adding this logic here is not quite what was intended, paths are supposed to be within the data directory.

For the rare case, mostly test cases, that you need to use a file as a resource there is the [Files.asResource( file )](https://github.com/geoserver/geoserver/blob/master/src/platform/src/main/java/org/geoserver/platform/resource/Files.java#L308) method.

The [Paths.convert( file, file )](https://github.com/geoserver/geoserver/blob/55c4740060eb2b8809ae935a510ac83eeb923313/src/platform/src/main/java/org/geoserver/platform/resource/Paths.java#L257) should be used rather than insert this logic?

Reply:
I disagree, if you don't do the above the file system resource will stop working on Windows, e.g., most methods stop working (think parent for example, it won't find the parent because it's looking for "/" while the path above in the constructor contains "\").

My guess is that the work was not tested against Windows at all, and probably not though out to work there too (the data directory does not live in the clouds, it normally comes from a real file system, if you get the path from File on Windows you end up with "\" in the path).

Can you try the whole thing on Windows so that you can check what works, what does not, adjust your expectations accordingly, and change the code so that it actually works? This is not my mess, someone else did it, I'm just trying to get a auto-complete to work.

Reply:
The code was definitely pushing into the constructor a "\" but I cannot replicate that anymore, and I've rewritten the file chooser lookup code so many times now that I don't remember which version managed to do that (I never called the constructor directly, there is some code path doing that by just using FileSystemResourceStore and Resource, but cannot find it right now). Removing it.

Reply:
I've also pushed a change to clarify in the javadocs that unix like paths need to be used, previously only one method did that.

Reply:
Thanks Andrea, the javadocs change is wise, I tried to use "path" consistently when a string is used, and File when a file is expected.

Your original stack trace, to check path validity, was intended to catch code that was passing in windows file system paths, or the old macos paths with ":" file system paths.